src/main/groovy/org/gradle/android/workarounds/RoomSchemaLocationWorkaround.groovy

```diff
@@ -223,6 +223,43 @@ class RoomSchemaLocationWorkaround implements Workaround {
             // maps annotation processor providers from the variant directly onto kapt tasks.
             javaCompileSchemaGenerationEnabled = false
         }
+
+
+        // Ksp Workaround implementation. Based on Kapt workaround
+        project.plugins.withId("com.google.devtools.ksp") {
+
+            // Ksp doesn't delegate to javac like kapt, we can use the task provider
+            project.tasks.withType(kspTaskClass).configureEach {
+
+                // Adding the Ksp CommandLineArgument with variant and room extension
+                def variantSpecificSchemaDir = project.objects.directoryProperty()
+                variantSpecificSchemaDir.set(getVariantSpecificSchemaDir(project, it.name))
+                it.commandLineArgumentProviders.add(new KspRoomSchemaLocationArgumentProvider(roomExtension.schemaLocationDir, variantSpecificSchemaDir))
+
+                def commandLineArgumentProviders = getAccessibleField(it.class, "__commandLineArgumentProviders__").get(it)
```

Comment:
why are we accessing this via reflection? isn't this the same as `it.commandLineArgumentProviders`?